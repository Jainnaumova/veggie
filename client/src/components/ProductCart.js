import React from 'react';

export default props => {
  const {product} = props;

  return (
    <div>
      <div className="productName">
        <h3>{product.name}</h3>
      </div>
      <img className="productImg" src={product.imageUrl} alt='product'/>
    </div>
  )
}
