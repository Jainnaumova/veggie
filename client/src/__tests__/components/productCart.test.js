import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';

import ProductCart from '../../components/ProductCart';

Enzyme.configure({ adapter: new Adapter() });

let props = {
  product: {
    name: 'Broccoli',
    imageUrl:
    'https://www.producemarketguide.com/sites/default/files/Commodities.tar/Commodities/broccoli_commodity-page.png'
  }
}

const component = () => shallow(<ProductCart {...props} />);

describe('ProductCart', () => {

  it('should match the snapshot', () => {
    const tree = renderer.create(
      // <MemoryRouter>
        <ProductCart {...props} />
      // </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should dynamic render ProductCart', () => {
    // change product
    props.product.name = 'Carrot';
    props.product.imageUrl = 'https://purepng.com/public/uploads/large/purepng.com-carrotscarrotvegetablesfreshdeliciousefoodhealthycarrots-481521740717jmglq.png';
    expect(component().find("h3").text()).toEqual("Carrot");
    // reset product
    props.product.name = 'Broccoli';
    props.product.imageUrl = 'https://www.producemarketguide.com/sites/default/files/Commodities.tar/Commodities/broccoli_commodity-page.png';
  });

  it("renders the name of the product in an <h3>", () => {
    expect(component().find("h3").text()).toEqual("Broccoli");
  });

})
