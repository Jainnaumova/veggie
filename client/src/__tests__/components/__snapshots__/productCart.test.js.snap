// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCart should match the snapshot 1`] = `
<div>
  <div
    className="productName"
  >
    <h3>
      Broccoli
    </h3>
  </div>
  <img
    alt="product"
    className="productImg"
    src="https://www.producemarketguide.com/sites/default/files/Commodities.tar/Commodities/broccoli_commodity-page.png"
  />
</div>
`;
